dali/operators/generic/reduce/reduce.h

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_REDUCE_REDUCE_H__
+#define DALI_OPERATORS_GENERIC_REDUCE_REDUCE_H__
+
+#include <vector>
+#include <algorithm>
+
+#include "dali/pipeline/operator/operator.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/reduce/reductions.h"
+#include "dali/kernels/reduce/reduce_cpu.h"
+
+namespace dali {
+#define REDUCE_TYPES (int16_t, int32_t, float)
+
+template <template <typename T, typename R> class ReductionType>
+class ReduceCPU : public Operator<CPUBackend> {
+ public:
+  explicit inline ReduceCPU(const OpSpec &spec) :
+    Operator<CPUBackend>(spec),
+    axes_(spec.GetRepeatedArgument<int>("axes")),
+    keep_dims_(spec.GetArgument<bool>("keep_dims")) { }
+
+  bool CanInferOutputs() const override { return true; }
+
+  inline ~ReduceCPU() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(ReduceCPU);
```

Comment:
I don't think this is necessary - isn't the base class marked as such?

Reply:
Done